core/primitives/src/crypto.rs

```diff
@@ -254,9 +255,10 @@ pub enum PublicError {
 }
 
 /// Key that can be encoded to/from SS58.
-#[cfg(feature = "std")]
+#[cfg(feature = "with_crypto")]
 pub trait Ss58Codec: Sized {
 	/// Some if the string is a properly encoded SS58Check address.
+	#[cfg(feature = "std")]
```

Comment:
Why is this only available with `std` feature?

Reply:
true, should be safe to enable

Reply:
In the end I get trouble becase I'd need to add `Mutex`:
https://github.com/paritytech/substrate/blob/48e6e349cd4d64fb94ea476089f9e4f900f31cf3/core/primitives/src/crypto.rs#L329

Reply:
I don't need this, so I can live with `#[cfg(feature = "std")]`

Reply:
Okay